crates/core/tedge/src/cli/certificate/cli.rs

```diff
@@ -70,6 +74,12 @@ impl BuildCommand for TEdgeCertCli {
                 };
                 cmd.into_boxed()
             }
+            TEdgeCertCli::Renew => {
+                let cmd = RenewCertCmd {
+                    cert_path: config.device.cert_path.clone(),
+                };
```

Comment:
Might be simpler to simply recreate also the private key if not required.

```suggestion
                let cmd = CreateCertCmd {
                    id: config.device.id.try_read(&config)?.to_string(),
                    cert_path: config.device.cert_path.clone(),
                    key_path: config.device.key_path.clone(),
                };
```

The error message behind the `?` needs to be improved to make it clear to the user that a certificate can only be renewed if already created.

Reply:
Just for clarification, is the private key recreated or is it only the public certificate which is recreated?